### ISSUE TITLE
Interoperability of persisted models

### ISSUE THREAD

**Issue by marcelklehr** [RubixML/ML#58]
Hey!
Thanks for what seems to be a huge amount of work. I appreciate it :)

I was wondering if it is possible to use models trained with a different framework? E.g. there is a large amount of pre-trained tensorflow models available and it would be amazing to be able to use those natively in PHP.

**Comment by andrewdalpino** [comment]
Hi @marcelklehr thanks for the great question

In fact we do have plans for interoperability, however, in the other direction

The [ONNX](https://onnx.ai/) format (or similar format), is a format that lets you serialize model data  such that you can train models at the comfort of your desk in PHP and then deploy the model on specialized inference systems (including TPUs, GPUs, and FPGAs) that speak ONNX. There are already a number of libraries in other languages that support this format, and Rubix ML would be the first in PHP.

In addition to being able to serve Rubix ML models in real-time with the [Server](https://github.com/RubixML/Server) library for normal volumes of traffic, you'd also be able to do real-time inference on massive amounts of traffic on specialized hardware if the need arises.

This is one of our top priorities

Thanks again for the great question, and I'll be happy to answer any followups

**Comment by marcelklehr** [comment]
Is there an ETA for this?

**Comment by andrewdalpino** [comment]
> Is there an ETA for this?

Nope, need more resources! However we are experimenting with building a library on top of TensorFlow called [Rubix DL](https://github.com/RubixML/DL). That might interest you.

**Comment by andrewdalpino** [comment]
Just an update because I know this feature means a lot to some people. Due to financial reasons, I had to drastically reduce the amount of time I have to work on Rubix-related projects. Unfortunately, this means that ONNX integration will be on hold for the foreseeable future. I invite others from the community to step up and contribute as this would be a huge step forward for the PHP ecosystem.

P.S. The library Rubix DL as mentioned earlier in this thread is also on hold pending more contributors or financial support from the community.
